layers/+lang/typescript/funcs.el

```diff
@@ -65,3 +65,19 @@
                  (list (point-min) (point-max))))
   (browse-url (concat "http://www.typescriptlang.org/Playground#src="
                       (url-hexify-string (buffer-substring-no-properties start end)))))
+
+(defun spacemacs//setup-typescript-mode ()
+  ;; Enable js-mode snippets
+  (yas-activate-extra-mode 'js-mode)
+  ;; See https://github.com/syl20bnr/spacemacs/issues/8222
+  (set (make-local-variable 'company-minimum-prefix-length) 2))
+
+(defun spacemacs//setup-typescript-tsx-mode ()
+  ;; Enable js-mode snippets
+  (yas-activate-extra-mode 'js-mode)
```

Comment:
good idea, didn't know this was a thing, but this should probably be hooked separately in `typescript/post-init-yasnippet`, otherwise this will error if someone has yasnippet excluded.

Reply:
I'm trying to do it inside the post init:
```
(defun typescript/post-init-yasnippet ()
  (yas-activate-extra-mode 'js-mode))
```
but yas-activate-extra-mode is void at that time

Reply:
```elisp
(defun typescript/post-init-yasnippet ()
  (with-eval-after-load 'yasnippet
    (yas-activate-extra-mode 'js-mode)))
```

Reply:
err ignore that

Reply:
I meant to add a hook within post-init-yasnippet. You don't want to activate the extra mode directly in there. This is mine:

```elisp
(defun aj-typescript/post-init-yasnippet ()
  (add-hook 'typescript-mode-hook #'aj-typescript//activate-extra-snippets)
  (add-hook 'typescript-tsx-mode-hook #'aj-typescript//activate-extra-snippets))

(defun aj-typescript//activate-extra-snippets ()
  (yas-activate-extra-mode 'js-mode))
```

Reply:
Also be sure to add yasnippet to the packages list